aten/src/ATen/native/Math.h

```diff
@@ -518,57 +518,62 @@ inline C10_HOST_DEVICE scalar_t calc_polygamma(scalar_t x, int n) {
  * gammainc, Cephes's igam and igamc, and Boost's Lanczos approximations.
  * See NOTICE for the licenses.
  */
-template <typename scalar_t>
-scalar_t ratevl(scalar_t x, const scalar_t num[], int64_t M,
+template <typename scalar_t, bool is_cuda=false>
+C10_HOST_DEVICE scalar_t
+ratevl(scalar_t x, const scalar_t num[], int64_t M,
     const scalar_t denom[], int64_t N) {
   // evaluating rational function, i.e., the ratio of two polynomials
   // the coefficients for numerator are given by `num` while coeffs for
-  // denumerator are given by `denom`
-
-  int64_t i, dir;
-  scalar_t y, num_ans, denom_ans;
-  scalar_t absx = std::fabs(x);
+  // denominator are given by `denom`
+  using acc_t = at::acc_type<scalar_t, is_cuda>;
```

Comment:
This is a bit outdated, you can use `at::opmath_type<scalar_t>` instead and you won't have to pass in `is_cuda` everywhere.